### PR TITLE
Using Django views instead of custom pages to handle responses.

### PR DESCRIPTION
**Pull request by saadyousafarbi**
Removing the extra failure.html and success.html and replacing them with built-in django templates. Also introducing dashboard.html where user is redirected after successful login.

### CODE REVIEW

File: autoshop/autoshop/settings.py [hunk 1]
```diff
@@ -23,9 +23,9 @@
 SECRET_KEY = 'i%&#7%nq24s#4xp+4b)w@$h)cy&nwotlz&pzb*s#kivh_6f#j6'
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = True
+DEBUG = False
```
Comment (zubair-arbi) [inline autoshop/autoshop/settings.py]: This is the right choice. But for local development we need to run django in debug mode. Please create a `private.py` settings file and override this value there

File: autoshop/core/templates/dashboard.html [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 {% block content %}
 <h1>
-    SUCCESS
+    Dashboard
```
Comment (zubair-arbi) [inline autoshop/core/templates/dashboard.html]: Remove this file all together. Redirect the user to `index` page which will serve as `homepage`.

File: autoshop/core/urls.py [hunk 3]
```diff
@@ -9,8 +9,7 @@
 urlpatterns = [
     url(r'^$', views.index, name='index'),
     url(r'^register/$', views.register, name='register'),
-    url(r'^failure/$', views.failure, name='failure'),
-    url(r'^success/$', views.success, name='success'),
+    url(r'^dashboard/$', views.dashboard, name='dashboard'),
```
Comment (zubair-arbi) [inline autoshop/core/urls.py]: No need

File: autoshop/core/views.py [hunk 4]
```diff
@@ -52,7 +54,7 @@ def register(request):
         print 'Profile for user "%s" failed to save due to validation errors: %s' % (
             request.user.username, form.errors
         )
-        return render(request, 'failure.html')
+        return Http404('Form is invalid:' + form.errors)
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: Revert this. We will redirect the user to profile edit page with form error message in our separate story.

File: autoshop/core/views.py [hunk 5]
```diff
@@ -74,10 +76,10 @@ def log_in(request):
                 password=form.data['password'],
             )
             if not user:
-                return render(request, 'failure.html')
+                raise PermissionDenied
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: Again we will redirect the user to login page with error message. We can do that with django messages.

File: autoshop/core/views.py [hunk 6]
```diff
@@ -163,6 +158,6 @@ def edit_profile(request):
         print 'Profile for user "%s" failed to save due to validation errors: %s' % (
             request.user.username, form.errors
         )
-        return render(request, 'failure.html')
+        return Http404('Form is invalid:' + form.errors)
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: 404 is for `page not found` error. Redirect user to previous page

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Can you please read the django docs on `Built-in` views again? https://docs.djangoproject.com/en/1.11/ref/views/#module-django.views
We need to introduce 404 and 500 pages with our own site wise styling. Please note the parameters here:
```
defaults.page_not_found(request, exception, template_name='404.html')

defaults.server_error(request, template_name='500.html')
```

File: autoshop/core/templates/login.html [hunk 7]
```diff
@@ -1,11 +1,24 @@
...
-    {% csrf_token %}
-    <table>
-        {{ form.as_table }}
-    </table>
-    <button type="submit" value="Submit" class="submit"> Sign in</button>
-</form>
+
+{% if messages %}
+<div class="span12">
+{% for message in messages %}
```
Comment (zubair-arbi) [inline autoshop/core/templates/login.html]: indentation

File: autoshop/core/urls.py (hunk 3 above)
Comment (zubair-arbi) [reply autoshop/core/urls.py]: remove this page as our requirement for now is to redirect user to homepage, i.e. `index`.

File: autoshop/core/views.py [hunk 8]
```diff
@@ -91,19 +95,12 @@ def log_out(request):
 
 
 @login_required()
-def success(request):
+def dashboard(request):
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: remove this view

File: autoshop/core/views.py [hunk 9]
```diff
@@ -163,6 +160,7 @@ def edit_profile(request):
         print 'Profile for user "%s" failed to save due to validation errors: %s' % (
             request.user.username, form.errors
         )
-        return render(request, 'failure.html')
+        messages.error(request, 'Form is invalid'+form.errors)
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: nit: spaces around the operator

File: autoshop/core/views.py [hunk 10]
```diff
@@ -52,7 +53,8 @@ def register(request):
         print 'Profile for user "%s" failed to save due to validation errors: %s' % (
             request.user.username, form.errors
         )
-        return render(request, 'failure.html')
+        messages.error(request, 'Form is invalid. Please try again.'+form.errors)
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: nit: spaces around operator

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Please add `404.html` and `500.html`

File: autoshop/core/templates/dashboard.html [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+        {{ message|safe }}
+    </div>
+{% endfor %}
+</div>
+{% endif %}
+
+<h1>
+    Dashboard
+</h1>
+<a href="/profile">profile</a>
```
Comment (zubair-arbi) [inline autoshop/core/templates/dashboard.html]: always use reverse urls.

**Review by zubair-arbi — APPROVED** [review]
Nice work 👍

**Comment by saadyousafarbi** [comment]
Rebased from e8075809ea6d861f8f44d0b1ca68103782a2c090

**Pull request merged**